src/common/actions/repository-input.js

```diff
@@ -0,0 +1,84 @@
+import 'isomorphic-fetch';
+import parseGitHubUrl from 'parse-github-url';
+
+export const CHANGE_REPOSITORY_INPUT = 'CHANGE_REPOSITORY_INPUT';
+export const UPDATE_STATUS_MESSAGE = 'UPDATE_STATUS_MESSAGE';
+export const SET_GITHUB_REPOSITORY = 'SET_GITHUB_REPOSITORY';
+export const VERIFY_GITHUB_REPOSITORY = 'VERIFY_GITHUB_REPOSITORY';
+export const VERIFY_GITHUB_REPOSITORY_SUCCESS = 'VERIFY_GITHUB_REPOSITORY_SUCCESS';
+export const VERIFY_GITHUB_REPOSITORY_ERROR = 'VERIFY_GITHUB_REPOSITORY_ERROR';
+
+export function updateInputValue(value) {
+  return (dispatch) => {
+    dispatch(changeRepositoryInput(value));
+    dispatch(validateGitHubRepository(value));
+  };
+}
+
+export function changeRepositoryInput(value) {
+  return {
+    type: CHANGE_REPOSITORY_INPUT,
+    payload: value
+  };
+}
+
+export function setGitHubRepository(repository) {
+  return {
+    type: SET_GITHUB_REPOSITORY,
+    payload: repository
+  };
+}
+
+export function validateGitHubRepository(repository) {
+  return (dispatch) => {
+    const gitHubRepo = parseGitHubUrl(repository);
```

Comment:
I wonder if this should be moved to the reducer? And that might help my argument on the number of and naming of actions to set the url?